presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -36,6 +39,42 @@
     protected MaterializedViewPlanValidator()
     {}
 
+    @Override
+    protected Void visitQuerySpecification(QuerySpecification node, MaterializedViewPlanValidatorContext context)
+    {
+        process(node.getSelect(), context);
+        if (node.getFrom().isPresent()) {
+            process(node.getFrom().get(), context);
+        }
+        if (node.getWhere().isPresent()) {
+            Expression subquery = node.getWhere().get();
+            boolean wasWithinWhere = context.isWithinWhere();
+
+            context.setWithinWhere(true);
```

Comment:
This might be a bug. We don't check subqueries in query specification. Subqueries can happen in multiple places. `visitSubqueryExpression` might be the right place to visit?

Reply:
Pls address this question. Both approaches look correct to me, after checking AstBuilder. visitSubqueryExpression might be more illustrative. Up to you.

Reply:
Pls address this question. Both approaches look correct to me, after checking AstBuilder. visitSubqueryExpression might be more illustrative. Up to you.

Reply:
subqueryExpression contains a Query which we will visit after that, so both work correctly.

Reply:
Is subquery something we want to prevent in general or not? What if it's in `HAVING` clause? Or what if we have it in other expressions like single row expression?

Reply:
We don't have to have this method. Just use
```java
    @Override
    protected Void visitTableSubquery(TableSubquery node, MaterializedViewPlanValidatorContext context)
    {
        // set within subquery to true
        process(node.getQuery(), context);
        // set outside subquery to false
        return null;
    }
```

f you are within a subquery, vistitTable will just throw

Reply:
Hi @pranjalssh let's hold on and discuss tomorrow.

Reply:
@highker valid point, missed some cases. Just wanted to allow subqueries within FROM clause. Let me check the sql grammar and fix this